server/src/test/java/org/apache/druid/server/http/CoordinatorCompactionConfigsResourceTest.java

```diff
@@ -24,6 +24,7 @@
 import org.apache.druid.audit.AuditManager;
 import org.apache.druid.common.config.ConfigManager;
 import org.apache.druid.common.config.JacksonConfigManager;
+import org.apache.druid.indexer.CompactionEngine;
```

Comment:
This test class seems to have no assertions on the newly added engine field. Please add/modify some tests that verify the engine value of the created config when engine is passed as `null` and when engine is explicitly passed as `native` or `msq`.

Reply:
There should also be a couple of tests that verify the exception and message that is thrown when an invalid config is passed. All the validation cases need not be captured as we are already doing that in `ClientCompactionRunnerInfoTest`.

Reply:
Augmented existing tests and added new ones.